ProcessLib/InitialCondition.h

```diff
@@ -10,38 +10,75 @@
 #ifndef PROCESS_LIB_INITIAL_CONDITION_H_
 #define PROCESS_LIB_INITIAL_CONDITION_H_
 
-#include <boost/property_tree/ptree.hpp>
-#include "logog/include/logog.hpp"
+#include <cassert>
+#include <boost/property_tree/ptree_fwd.hpp>
+#include "MeshLib/Node.h"
+#include "MeshLib/PropertyVector.h"
 
-#include "MeshLib/Mesh.h"
+namespace MeshLib
+{
+template <typename>
+class PropertyVector;
+class Mesh;
+}
 
 namespace ProcessLib
 {
-
+/// The InitialCondition is a base class for spatial distributions of values
+/// defined on mesh nodes.
 class InitialCondition
 {
 public:
-    virtual ~InitialCondition() = default;
+	virtual ~InitialCondition() = default;
+	virtual double getValue(MeshLib::Node const&) const = 0;
 };
 
-
+/// Uniform value initial condition
 class UniformInitialCondition : public InitialCondition
 {
-    using ConfigTree = boost::property_tree::ptree;
 public:
-    UniformInitialCondition(ConfigTree const& config)
-    {
-        DBUG("Constructing Uniform initial condition");
+	UniformInitialCondition(double const value) : _value(value)
+	{
+	}
+
+	virtual double getValue(MeshLib::Node const&) const override
+	{
+		return _value;
+	}
+
+private:
+	double _value;
+};
+
+using ConfigTree = boost::property_tree::ptree;
```

Comment:
Can that interfere with other definitions of ConfigTree? At least, there should be no different definitions, but maybe the compiler could complain?